src/components/icon/icon.component.js

```diff
@@ -1,65 +1,70 @@
 import React from "react";
 import PropTypes from "prop-types";
-import TooltipDecorator from "../../utils/decorators/tooltip-decorator";
-import { validProps } from "../../utils/ether";
 import tagComponent from "../../utils/helpers/tags";
 import StyledIcon from "./icon.style";
 
-class Icon extends React.Component {
-  /** Return component props */
-  get componentProps() {
-    return validProps(this);
-  }
-
-  /** Return Icon type with overrides */
-  get type() {
-    // switch tweaks icon names for actual icons in the set
-    switch (this.props.type) {
-      case "help":
-        return "question";
-      case "maintenance":
-        return "settings";
-      case "new":
-        return "gift";
-      case "success":
-        return "tick";
-      case "messages":
-      case "email":
-        return "message";
-      default:
-        return this.props.type;
-    }
-  }
+const Icon = React.forwardRef(
+  (
+    {
+      bg,
+      bgShape,
+      bgSize,
+      bgTheme,
+      className,
+      color,
+      disabled,
+      fontSize,
+      iconColor,
+      type,
+      ml,
+      mr,
+      ...props
+    },
+    ref
+  ) => {
+    /** Return Icon type with overrides */
+    const iconType = () => {
+      // switch tweaks icon names for actual icons in the set
+      switch (type) {
+        case "help":
+          return "question";
+        case "maintenance":
+          return "settings";
+        case "new":
+          return "gift";
+        case "success":
+          return "tick";
+        case "messages":
+        case "email":
+          return "message";
+        default:
+          return type;
+      }
+    };
 
-  /** Renders the component. */
-  render() {
-    return [
+    return (
       <StyledIcon
-        bgSize={this.props.bgSize}
-        bgShape={this.props.bgShape}
-        bgTheme={this.props.bgTheme}
-        fontSize={this.props.fontSize}
-        iconColor={this.props.iconColor}
-        disabled={this.props.disabled}
-        color={this.props.color}
-        bg={this.props.bg}
-        type={this.type}
+        ref={ref}
+        bgSize={bgSize}
+        bgShape={bgShape}
+        bgTheme={bgTheme}
+        fontSize={fontSize}
+        iconColor={iconColor}
+        disabled={disabled}
+        color={color}
+        bg={bg}
+        type={iconType()}
         key="icon"
-        className={this.props.className || null}
-        {...this.componentProps}
-        {...tagComponent("icon", this.props)}
-        ref={(comp) => {
-          this._target = comp;
-        }}
-        data-element={this.type}
-        mr={this.props.mr}
-        ml={this.props.ml}
-        aria-label={this.props.ariaLabel}
-      />,
-      this.tooltipHTML,
-    ];
+        className={className || null}
+        {...tagComponent("icon", props)}
+        data-element={iconType()}
+        mr={mr}
+        ml={ml}
+        {...props}
```

Comment:
Do we really want to pass all the rest of the props down to `StyledIcon` here? Anything that `StyledIcon` needs should be passed down explicitly and nothing else should be passed to it - it'll help avoid problems like we're having with `FormField` etc at the moment

Reply:
Yeah good point I'll sort it once the reviews have finished